src/DebuggerForm.cpp

```diff
@@ -1476,15 +1481,17 @@ void DebuggerForm::symbolFileChanged()
 DebuggerForm::AddressSlotResult DebuggerForm::addressSlot(int addr) const
 {
 	int p = (addr & 0xC000) >> 14;
-	auto ps = qint8(memLayout.primarySlot[p]);
-	auto ss = qint8(memLayout.secondarySlot[p]);
-	int segment = [&] { // figure out (rom) mapper segment
-		if (memLayout.mapperSize[ps][ss == -1 ? 0 : ss] > 0) {
-			return memLayout.mapperSegment[p];
+	uint8_t ps = memLayout.primarySlot[p];
+	int8_t ss_ = memLayout.secondarySlot[p];
+	auto ss = make_positive_optional<uint8_t>(ss_);
+
+	std::optional<uint8_t> segment = [&] { // figure out (rom) mapper segment
+		if (ss && memLayout.mapperSize[ps][*ss] > 0) {
+			return std::optional((uint8_t) memLayout.mapperSegment[p]);
 		} else {
 			int q = 2 * p + ((addr & 0x2000) >> 13);
 			int b = memLayout.romBlock[q];
-			return (b >= 0) ? b : -1;
+			return make_if(b >= 0, (uint8_t) b);
```

Comment:
detail: line 1490 and 1494: prefer function-style cast over C-style-cast. So `uint8_t(b)` instead of `(uint8_t)b`.